### PR TITLE
ACS-3832: test support for parent queries against categories

### PR DESCRIPTION
**Pull request by Pawel-608**

### CODE REVIEW

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/NodesSecondaryChildrenRelatedTests.java [hunk 1]
```diff
@@ -211,6 +219,8 @@ protected FileModel createDocument(String filename)
                 .createContent(new FileModel(filename, FileType.TEXT_PLAIN, "content"));
         }
 
+
+
```
Comment (tpage-alfresco) [inline tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/NodesSecondaryChildrenRelatedTests.java:223]: Random extra new lines added :rocket: 

We should look into automatic formatting :slightly_smiling_face:

File: tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/utils/Categories.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Map;
+
+import org.alfresco.rest.core.RestWrapper;
+import org.alfresco.rest.model.RestCategoryModel;
+import org.alfresco.utility.data.DataUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Categories
```
[the remark this answers is not in the document]
Comment (krdabrowski) [reply tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/utils/Categories.java:16]: I agree, we should move this stuff into comm-repo - we could do it in code review if there will be enough of time, or in tech debt week.

[the remark this answers is not in the document]
Comment (Pawel-608) [reply tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/utils/Categories.java:16]: I would move it there in tech debt week

**Review by mpichura — APPROVED** [review]
LGTM

**Pull request merged**